backend/src/app.controller.ts

```diff
@@ -0,0 +1,47 @@
+import { Body, Controller, Get, HttpCode, Post, Query } from '@nestjs/common';
+import { Node } from './entities/Node';
+import { Edge } from './entities/Edge';
+import { ParseIntArrayPipe } from './pipes/ParseIntArrayPipe';
+import { AppService } from './app.service';
+import { LimitQuery } from './entities/queries/LimitQuery';
+import { QueryResult } from './entities/queries/QueryResult';
+
+/**
+ * Main App Controller.
+ * ATM there are no only a few endpoints, so only one controller is used.
+ */
+@Controller()
+export class AppController {
+  constructor(private readonly appService: AppService) {}
+
+  /**
+   * Returns a list the ids of all nodes
+   */
+  @Post('queryAll')
+  @HttpCode(200)
+  async queryAll(@Body() query?: LimitQuery): Promise<QueryResult> {
+    return this.appService.queryAll(query);
+  }
+
+  /**
+   * Returns list of Nodes
```

Comment:
Maybe describe this in more details, that you get a batch of node details for the node ids that are part of the request.
Maybe also add the call signature, like is done for `getEdgesById`